base/quadgk.jl

```diff
@@ -169,10 +169,10 @@ end
 # all the integration-segment endpoints
 function quadgk(f, a, b, c...; kws...)
     T = promote_type(typeof(float(a)), typeof(b))
-    for i in 1:length(c)
-        T = promote_type(T, typeof(c[i]))
+    for x in c
+        T = promote_type(T, typeof(x))
     end
-    cT = T[ c[i] for i in 1:length(c) ]
+    cT = map(T, c)
```

Comment:
Here it should be fine; you know `c` is a tuple, and indeed the tuple output seems like the better choice.